src/models/SpectralDefinition.ts

```diff
@@ -124,3 +139,229 @@ export const SPECTRAL_COORDS_SUPPORTED = new Map<string, {type: SpectralType; un
+            options.push(IntensityUnitType.Kelvin);
+            if (isFinite(config?.bmaj) && isFinite(config?.bmin) && isFinite(config?.freqGHz)) {
+                options.push(IntensityUnitType.JyBeam);
+                options.push(IntensityUnitType.JySr);
+                options.push(IntensityUnitType.JyArcsec2);
+            }
+        } else if (type === IntensityUnitType.JyBeam) {
+            options.push(IntensityUnitType.JyBeam);
+            if (isFinite(config?.bmaj) && isFinite(config?.bmin)) {
+                options.push(IntensityUnitType.JySr);
+                options.push(IntensityUnitType.JyArcsec2);
+                if (isFinite(config?.freqGHz)) {
+                    options.push(IntensityUnitType.Kelvin);
+                }
+            }
+        } else if (type === IntensityUnitType.JySr || type === IntensityUnitType.JyArcsec2) {
+            options.push(IntensityUnitType.JySr);
+            options.push(IntensityUnitType.JyArcsec2);
+            if (isFinite(config?.bmaj) && isFinite(config?.bmin)) {
+                options.push(IntensityUnitType.JyBeam);
+                if (isFinite(config?.freqGHz)) {
+                    options.push(IntensityUnitType.Kelvin);
+                }
+            }
+            if (isFinite(config?.cdelta1) && isFinite(config?.cdelta2)) {
+                options.push(IntensityUnitType.JyPixel);
+            }
+        } else if (type === IntensityUnitType.JyPixel) {
+            options.push(IntensityUnitType.JyPixel);
+            if (isFinite(config?.cdelta1) && isFinite(config?.cdelta2)) {
+                options.push(IntensityUnitType.JySr);
+                options.push(IntensityUnitType.JyArcsec2);
+            }
+        }
+    }
+    return options;
+};
+
+export const GetIntensityOptions = (config: IntensityConfig): string[] => {
+    const convertibleTypes = FindConvertibleIntensityTypes(config);
+    let supportedOptions = [];
+    convertibleTypes?.forEach(type => {
+        supportedOptions.push(...IntensityOptionsMap.get(type));
+    });
+    return supportedOptions;
+};
+
+const JyBeamToKelvin = (freqGHz: number, bmaj: number, bmin: number, forward: boolean = true): number => {
+    const coefficient = (1.222 * 1e6) / (freqGHz * freqGHz * bmaj * bmin);
+    return forward ? coefficient : 1 / coefficient;
+};
+
+const JyBeamToJySr = (bmaj: number, bmin: number, forward: boolean = true): number => {
+    const bmajRad = (bmaj * Math.PI) / 648000;
+    const bminRad = (bmin * Math.PI) / 648000;
+    const omega = (Math.PI * bmajRad * bminRad) / (4 * Math.LN2);
+    return forward ? 1 / omega : omega;
+};
+
+const JySrToJyArcsec2 = (forward: boolean = true): number => {
+    const constant = 2.350443 * 1e-11;
+    return forward ? constant : 1 / constant;
+};
+
+const JyPixelToJyArcsec2 = (cdelta1: number, cdelta2: number, forward: boolean = true): number => {
+    const coefficient = (cdelta1 * cdelta2) / (2.350443 * 1e-11);
```

Comment:
please check the units of cdelta1 and cdelta2. They should be in rad, not degree.